common/app/navigation/Navigation.scala

```diff
@@ -146,16 +146,28 @@ object NavMenu {
     s"/$id"
   }
 
-  private[navigation] def getSubnav(currentNavLink: Option[NavLink], currentParent: Option[NavLink], currentPillar: Option[NavLink]): Subnav = {
-    val currentNavHasChildren = currentNavLink.exists(_.children.nonEmpty)
-    val parentIsPillar =  currentParent.equals(currentPillar)
-    def currentNavIsPillar = currentNavLink.equals(currentPillar)
+  private[navigation] def getCustomSignPosting(navItem: NavItem): Subnav = {
+    val children = navItem.links.map(link => {
+      NavLink(link.breadcrumbTitle, link.href)
+    })
+    val parent = NavLink(navItem.name.breadcrumbTitle, navItem.name.href)
+    Subnav(Some(parent), children)
+  }
+
+  private[navigation] def getSubnav(page: Page, currentNavLink: Option[NavLink], currentParent: Option[NavLink], currentPillar: Option[NavLink]): Subnav = {
+    page.metadata.customSignPosting.map( customSignPosting => getCustomSignPosting(customSignPosting)).getOrElse {
+
+      val currentNavHasChildren = currentNavLink.exists(_.children.nonEmpty)
+      val parentIsPillar = currentParent.equals(currentPillar)
 
-    val parent = if(currentNavHasChildren & !currentNavIsPillar) currentNavLink else if (parentIsPillar) None else currentParent
+      def currentNavIsPillar = currentNavLink.equals(currentPillar)
 
-    val childrenToShow = if (currentNavHasChildren) currentNavLink else currentParent
-    val children = childrenToShow.map( navLink => navLink.children ).getOrElse(Nil)
+      val parent = if (currentNavHasChildren & !currentNavIsPillar) currentNavLink else if (parentIsPillar) None else currentParent
 
-    Subnav(parent, children)
+      val childrenToShow = if (currentNavHasChildren) currentNavLink else currentParent
+      val children = childrenToShow.map(navLink => navLink.children).getOrElse(Nil)
+
+      Subnav(parent, children)
+    }
```

Comment:
Rather than nesting the subnav creating logic in the `getOrElse` I would create an internal function or lazy val instead to make the `getOrElse` logic clearer. Ex:

```
private[navigation] def getSubnav(page: Page, currentNavLink: Option[NavLink], currentParent: Option[NavLink], currentPillar: Option[NavLink]): Subnav = {
   lazy val subNav = {
      val currentNavHasChildren = currentNavLink.exists(_.children.nonEmpty)
      val parentIsPillar =  currentParent.equals(currentPillar)
      lazy val currentNavIsPillar = currentNavLink.equals(currentPillar)
...
      Subnav(parent, children)
   }
   page.metadata.customSignPosting.map(getCustomSignPosting).getOrElse(subnav)
}
```

Reply:
Thanks for your helpful comments! All should be addressed now

Reply:
Thank you @NataliaLKB